config/configmodels/configmodels.go

```diff
@@ -44,15 +44,50 @@ type Config struct {
 type NamedEntity interface {
 	Name() string
 	SetName(name string)
+	Type() string
+	SetType(typeStr string)
+	IsEnabled() bool
+}
+
+// Entity implements NamedEntity.
+type Entity struct {
+	typeVal string `mapstructure:"-"`
+	nameVal string `mapstructure:"-"`
+	// Configures if the entity is disabled. The default value is false (meaning the entity is enabled by default).
+	disabled bool `mapstructure:"disabled"`
+}
+
+// Name gets the entity name.
+func (e *Entity) Name() string {
+	return e.nameVal
+}
+
+// SetName sets the exporter name.
+func (e *Entity) SetName(name string) {
+	e.nameVal = name
+}
+
+// Type sets the exporter type.
+func (e *Entity) Type() string {
+	return e.typeVal
+}
+
+// SetType sets the exporter type.
+func (e *Entity) SetType(typeStr string) {
+	e.typeVal = typeStr
 }
 
+// IsEnabled returns true if the entity is enabled.
+func (e *Entity) IsEnabled() bool {
+	return !e.disabled
+}
+
+var _ NamedEntity = (*Entity)(nil)
+
 // Receiver is the configuration of a receiver. Specific receivers must implement this
 // interface and will typically embed ReceiverSettings struct or a struct that extends it.
 type Receiver interface {
 	NamedEntity
-	IsEnabled() bool
-	Type() string
-	SetType(typeStr string)
```

Comment:
If we want to get rid of repetition here then probably better to define another interface and use it instead of modifying NamedEntity to what it is not. For example:
```go
type ComponentConfig interface {
    NamedEntity
    IsEnabled() bool
    Type() string
    SetType(typeStr string)
}
```
Or you can instead just rename `NamedEntity` to `ComponentConfig`.